### PR TITLE
WIP: If you liked it than you should have put a URL on it

### LINKED ISSUE #152: If you liked it then you should have put a URL on it.
Cc @adelevie

### PR DESCRIPTION
**Pull request by benbalter**
Fixes #152. /cc @adelevie

### CODE REVIEW

**Comment by benbalter** [comment]
@adelevie struggling a bit to get across two very different ideas:
1. Have URLs for stuff on the internet
2. Expose process through URLs (stuff not yet on the internet)

halp?

**Comment by adelevie** [comment]
At the DCLH event that @compleatang just spoke at. Will take a look soon!

File: _posts/2014-10-02-expose-process-through-urls.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
+---
+title: "If you liked it then you should have put a URL on it"
+excerpt: ""
+---
+
+If there's one thing that made the internet what it is, it's the URL. It's what makes sharing funny cat videos possible. It's the bookmark to access your bank account balance. It's the Wikipedia link you send your budy to end a heated argument. Why then, are our most important online interactions — collaborating with coworkers, accessing government services, or consuming open data — all too often, tragically URL-less?
```
Comment (parkr) [inline _posts/2014-10-02-expose-process-through-urls.md]: `buddy`

Comment (parkr) [inline _posts/2014-10-02-expose-process-through-urls.md]: i'd also argue that even _offline_ conversations and interactions deserve a URL

File: _posts/2014-10-02-expose-process-through-urls.md [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+
+## The humble URL
+
+There's a lot packed into that seemingly innocuous string at the top of your browser. At its most basic level, a URL is a promise that so long as they have that magical sequence of characters, anyone in the world can see exactly what you see. In technical terms, a URL or [Uniform Resource Locator](https://en.wikipedia.org/wiki/Uniform_resource_locator) is the means by which to access a resource. That resource can be anything from a product on Amazon.com to an individual Tweet on Twitter to a page in the Congressional Record. A URL is also, by definition, also a URI ([uniform resource identifier](Uniform_resource_identifier)), meaning no two things ever have the same URL. Put another way, if you have something's URL, you also have a unique way to identify it, today and in perpetuity.
```
Comment (parkr) [inline _posts/2014-10-02-expose-process-through-urls.md]: uri is missing the link: http://en.wikipedia.org/wiki/Uniform_resource_identifier

Comment (parkr) [inline _posts/2014-10-02-expose-process-through-urls.md]: obvious caveats to perpetuity to be addressed?

File: _posts/2014-10-02-expose-process-through-urls.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+## Exposing process
+
+But URLs are more than just the phone numbers or license plates for the internet. When used correctly, [they expose process](http://tomayko.com/writings/adopt-an-open-source-process-constraints). Whereas in many organizations, asking why something is the way it is, involves several trips up and down the hall to "ask Bob, I think he might have been around then" (with Bob somehow always being out of the office), at GitHub, as in most of open source, every decision has its own URL.
```
Comment (parkr) [inline _posts/2014-10-02-expose-process-through-urls.md]: Don't need the extra comma in the second sentence. `Whereas in many organizations, asking why something is the way it is involves several trips up and down the hall. At GitHub, as it is for most open source projects, every decision and every conversation has its own URL.`

File: _posts/2014-10-02-expose-process-through-urls.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+That URL exposes not just what decision was made, but why it was made, who made it, what information was available at the time, and what arguments were made both for and against it. And as new members onboard, they have immediate (and non-blocking) access to that URL, and all the context it contains. Better still, as the same or related issues arise, that institutional knowledge can be passed around with a single hyperlink.
```
Comment (parkr) [inline _posts/2014-10-02-expose-process-through-urls.md]: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse: :heart_eyes: :heartbeat: :heartpulse:

File: _posts/2014-10-02-expose-process-through-urls.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+
+We rarely use email for exactly that reason. It's not available to those not originally CC'd, it's not easily referenced, and most importantly, it's easily lost. What was the subject again? Was it sent to this DL or the other? Did I go over my quota? That medium doesn't matter (issues, forums, chat with transcripts) so long as it's possible to move backward from a given deliverable and understand how things came to be.
```
Comment (parkr) [inline _posts/2014-10-02-expose-process-through-urls.md]: Would be great to describe what `DL` is. I don't know what you mean by that.

File: _posts/2014-10-02-expose-process-through-urls.md [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+
+1. Go to [`sam.gov`](http://sam.gov)
+2. Click `Search Records`
+3. Type in the word "GitHub"
+4. Click `Search`
+5. Click `View Details`
+
+Sadly, that's the most direct way anyone can get to that public information. Take a look at your URL and you'll see things like `navigationalstate`, `interactionstate`, each followed by an obscure string of letters and numbers. This tells you that the URL is only part of the puzzle, and that the server decides what to display based primarily on your previous interactions. The server, not the URL, is the master of what you see.
+
+The instructions to how to retrieve content on the internet should not involve sub-instructions like "click", "scroll", or "search". "Take this URL" should be the only instruction you need. The internet is [stateless](https://en.wikipedia.org/wiki/Stateless_protocol),
```
Comment (parkr) [inline _posts/2014-10-02-expose-process-through-urls.md]: What do you think about

``` text
The instructions for how to retrieve content on the Internet should not involve more than one step.
Any instructions containing "click," "scroll," or "search" are a sure sign that we haven't reached URL nirvana.
Simply "Go to this URL," and you're there. The Internet is [stateless](https://en.wikipedia.org/wiki/Stateless_protocol).
```

File: _posts/2014-10-02-expose-process-through-urls.md [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+### The internet is not an afterthought
+
+## Great URLs
+
+Great URLs take that three steps further:
+
+  * **They're semantic** - URLs may be consumed by machines, but they are primarily built for humans. As such, [they should describe the thing they represent in human-readable, not machine-readable terms](https://en.wikipedia.org/wiki/Semantic_URL). We've all looked to the top of our screen when browsing, only to discover `domain.com/index.php?page_id=23784` or a similar obscure database ID. Great URLs make it immediately clear what they represent. Just by seeing `ben.balter.com/2014/10/02/expose-process-through-urls` you immediately know (A) that this is an article, (B) who the author of that article is (C) when it was published, and (D) what it's about.
```
Comment (parkr) [inline _posts/2014-10-02-expose-process-through-urls.md]: `should describe the thing they represent in human-readable, not just machine-readable terms` ?

File: _posts/2014-10-02-expose-process-through-urls.md [hunk 8]
```diff
@@ -0,0 +1,48 @@
...
+2. Click `Search Records`
+3. Type in the word "GitHub"
+4. Click `Search`
+5. Click `View Details`
+
+Sadly, that's the most direct way anyone can get to that public information. Take a look at your URL and you'll see things like `navigationalstate`, `interactionstate`, each followed by an obscure string of letters and numbers. This tells you that the URL is only part of the puzzle, and that the server decides what to display based primarily on your previous interactions. The server, not the URL, is the master of what you see.
+
+The instructions to how to retrieve content on the internet should not involve sub-instructions like "click", "scroll", or "search". "Take this URL" should be the only instruction you need. The internet is [stateless](https://en.wikipedia.org/wiki/Stateless_protocol),
+
+### The internet is not an afterthought
```
Comment (parkr) [inline _posts/2014-10-02-expose-process-through-urls.md]: Looking forward to reading this piece :)

**Comment by benbalter** [comment]
@parkr thanks for the feedback!

**Comment by parkr** [comment]
@benbalter my pleasure, hope it was helpful

**Comment by benbalter** [comment]
> hope it was helpful

Very. On spotty internet, so couldn't comment line-by-line, but believe I incorporated everything.

**Pull request merged**